phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -3286,6 +3288,43 @@ void createSysMutexTableIfNotExists(HBaseAdmin admin) throws IOException, SQLExc
         }
     }
 
+    /**
+     * Check if the SYSTEM MUTEX table exists. If it does, ensure that its TTL is correct and if
+     * not, modify its table descriptor
+     * @param admin HBase admin
+     * @return true if SYSTEM MUTEX exists already and false if it needs to be created
+     * @throws IOException thrown if there is an error getting the table descriptor
+     */
+    @VisibleForTesting
+    boolean checkIfSysMutexExistsAndModifyTTLIfRequired(HBaseAdmin admin) throws IOException {
+        HTableDescriptor htd;
+        try {
+            htd = admin.getTableDescriptor(Bytes.toBytes(SYSTEM_MUTEX_NAME));
+        } catch (org.apache.hadoop.hbase.TableNotFoundException ignored) {
+            try {
+                // Try with the namespace mapping name
+                htd = admin.getTableDescriptor(TableName.valueOf(SYSTEM_SCHEMA_NAME,
+                        SYSTEM_MUTEX_TABLE_NAME));
+            } catch (org.apache.hadoop.hbase.TableNotFoundException ignored2) {
+                return false;
+            }
+        }
+
+        // The SYSTEM MUTEX table already exists so check its TTL
+        if (htd.getFamily(SYSTEM_MUTEX_FAMILY_NAME_BYTES).getTimeToLive() != TTL_FOR_MUTEX) {
+            LOGGER.debug("SYSTEM MUTEX already appears to exist, but has the wrong TTL. " +
+                    "Will modify the TTL");
+            HColumnDescriptor hColFamDesc = htd.removeFamily(SYSTEM_MUTEX_FAMILY_NAME_BYTES);
+            hColFamDesc.setTimeToLive(TTL_FOR_MUTEX);
+            htd.addFamily(hColFamDesc);
+            admin.modifyTable(htd.getTableName(), htd);
```

Comment:
Does this reload the region/store, not sure offhand, but good to check, if yes then ignore my previous on comment about region restart/reload.